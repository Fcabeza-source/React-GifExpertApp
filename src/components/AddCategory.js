import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState("Ingrese la serie busqueda...");

    const handleInputChange = (e) => {
        setInputValue(e.target.value); 
    };

    const handleSubmit = (e) => {
        e.preventDefault();

        if ( inputValue.trim().length > 2){
            setCategories( cats => [inputValue, ...cats ]);
            setInputValue("");
        }
    };

    const handleErrase = (e) => {
            setInputValue(e.value="");
        }


    return (

        <form onSubmit={handleSubmit}>
            <input 
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                onClick={handleErrase}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories : PropTypes.func.isRequired ,
}
