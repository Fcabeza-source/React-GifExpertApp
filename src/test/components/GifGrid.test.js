import'@testing-library/jest-dom';
import React from 'react';

import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGIfs';
jest.mock('../../hooks/useFetchGIfs');




describe('Pruebas en <GifGrid />', () => {
     
    const category = 'One Piece';
    
    test('Debe de hacer match con la snapshot', () => {

        useFetchGifs.mockReturnValue({
            data : [] ,
            loading : true,
        });
        
        const wrapper = shallow(<GifGrid category = { category } />);
        
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar items cuandi se cargan imagenes UseFetchGrids ', () => {
        
        const gifs = [{
            id : 'ABC' ,
            url : 'https://jajduwioauoiud.jpg' , 
            title : 'Cualquier cosa ' 
        }]

        useFetchGifs.mockReturnValue({
            data : gifs ,
            loading : false
        });
        
        const wrapper = shallow(<GifGrid category = { category } />);

        expect(shallow).toMatchSnapshot();
        expect((wrapper).find('p').exists()).toBe(false);
        expect( wrapper.find('GifGridItem').length).toBe ( gifs.length );


    })
    
})
