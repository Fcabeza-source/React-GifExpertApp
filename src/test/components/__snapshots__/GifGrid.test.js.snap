// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> Debe de hacer match con la snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Piece
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en <GifGrid /> Debe de mostrar items cuandi se cargan imagenes UseFetchGrids  1`] = `[Function]`;
